### PR TITLE
Add User-Agent for gitea's self-implemented lfs client.

### PR DESCRIPTION
**Pull request by hiifong**
See https://github.com/go-gitea/gitea/pull/32812#issuecomment-2540593960

img_v3_02hh_00da7d2a-43f8-4b89-a18a-bdfa85e3693g

### CODE REVIEW

File: modules/lfs/shared.go [hunk 1]
```diff
@@ -14,11 +15,15 @@ import (
...
-	AcceptHeader    = "application/vnd.git-lfs+json;q=0.9, */*;q=0.8"
-	UserAgentHeader = "git-lfs"
+	// AcceptHeader Some LFS servers offer content with other types, so fallback to '*/*' if application/vnd.git-lfs+json cannot be served
+	AcceptHeader = "application/vnd.git-lfs+json;q=0.9, */*;q=0.8"
 )
 
+// UserAgentHeader Add User-Agent for gitea's self-implemented lfs client,
+// and the version is consistent with the latest version of git lfs can be avoided incompatibilities.
+// Some lfs servers will check this
+var UserAgentHeader = fmt.Sprintf("git-lfs/3.6.0 (Gitea; %s %s; %s)", runtime.GOOS, runtime.GOARCH, runtime.Version())
```
Comment (lunny) [inline modules/lfs/shared.go]: How about using Gitea's version instead of 3.6.0

Comment (wxiaoguang) [inline modules/lfs/shared.go]:
```suggestion
var UserAgentHeader = "git-lfs/3.6.0 (Gitea)"
```

I believe some users won't like to leak their "server information". And we do not need to call the Sprintf now (faster)

Comment (wxiaoguang) [reply modules/lfs/shared.go]: NO. The comment is clear enough: it is "git lfs version"

File: modules/lfs/shared.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ package lfs
 import (
 	"errors"
 	"fmt"
+	"runtime"
```
Comment (wxiaoguang) [inline modules/lfs/shared.go]:
```suggestion
```

**Review by wxiaoguang — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**